docs/blog/20230903-confused-Techie-pulsars-ci.md

```diff
@@ -0,0 +1,141 @@
+
+Now, I don't want to come across as ungrateful to the fantastic service that Cirrus offers, so when this price change was first announced, we had originally intended to figure out if we could afford these price changes reasonably.
+
+To determine what we could afford, we first had to estimate what we would've been paying the past months of usage. [@DeeDeeG] was able to break down exactly what our costs looked like, using our previous months data.
+
+Keeping in mind that we trigger the same amount of runs per platform, the difference here is how slow or fast any given platform is during the build process:
+
+| Month | Windows Credits/Hours  | Linux Credits/Hours    | macOS Credits/Hours     |
+| ----- | ---------------------- | ---------------------- | ----------------------- |
+| May   | 66 credits / 273 hours | 72credits / 396 hours  | 490 credits / 544 hours |
+| July  | 55 credits / 229 hours | 39 credits / 213 hours | 305 credits / 339 hours |
+
+Using this data, where we consider 1 credits = $1 USD, minus the novel 50 free credits, July would've costed $349 USD. We could then realistically estimate that building Pulsar binaries would cost ~$350 USD a month.
+
+Which when we consider that's a higher price than anything the Pulsar team spends any money on for Pulsar, it's a bit of a tough pill to swallow. But especially because Cirrus isn't the only service out there, no one on the team could justify spending our sponsors money on such a large expense. At this point, we had to determine a way to lower our usage or move off the platform.
+
+But to move off Cirrus, we had to consider everything it currently does for us:
+
+- Builds binaries on every Pull Request
+- Builds our Regular Release Binaires
+- Builds our Rolling Release Binaries (on every commit to `master`)
+- Acts as a CDN for our Rolling Release Binaries (The `download` microservice redirects downloads directly to Cirrus)
+
+At this point, with the above knowledge, we had several different solutions to jugle and consider:
+
+- We could move to another CI platform, but Cirrus is one of the cheaper ones
+- We could move most of our process to GitHub Actions
+- We could move everything to non-CI services, such as cloud hosted VMs
+- We could store binaires in other cloud storage services, such as GCP or Digital Ocean
+- We could slim down our Cirrus usage as much as possible
+
+But with all of these ideas, as a Do-ocracy, [attempts](https://github.com/pulsar-edit/pulsar/pull/682) began being made to switch as much as possible to GitHub Actions.
+
+While this did solve a few of our needs from above, it didn't solve all of them, and introduced it's own issues.
+
+The biggest one, GitHub Actions doesn't support Apple Silicon or ARM Linux. Even though Apple Silicon support is on GitHub's [Roadmap](https://github.com/orgs/github/projects/4247), it has been pushed forward over and over since June 15th, 2022. So this support wasn't something we could rely on.
```

Comment:
```suggestion
The biggest one: GitHub Actions doesn't support Apple Silicon or ARM Linux. Even though Apple Silicon support is on GitHub's [Roadmap](https://github.com/orgs/github/projects/4247), it has been pushed forward over and over since June 15th, 2022. So this support wasn't something we could rely on.
```